Jeremy Chia iP

jeremychiaaaaa opened the pull request:
### NextGPT

NextGPT frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use
- All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features:

- [X]  Managing tasks
- [X]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jeremychiaaaaa committed:
Changed default bot name + renamed file + welcome & exit message

jeremychiaaaaa committed:
Level-1

jeremychiaaaaa committed:
Level 2

jeremychiaaaaa committed:
Level-3

jeremychiaaaaa committed:
Level-4

jeremychiaaaaa committed:
A-TextUiTesting

jeremychiaaaaa committed:
Level-5

jeremychiaaaaa committed:
Level-6

jeremychiaaaaa committed:
Add Level 7

jeremychiaaaaa committed:
Add Level-8

jeremychiaaaaa committed:
Add A-MoreOOP

jeremychiaaaaa committed:
Add A-Packages

jeremychiaaaaa committed:
Add A-Gradle

jeremychiaaaaa committed:
Add A-Gradle

jeremychiaaaaa committed:
Add J-Unit tests

jeremychiaaaaa committed:
Add A-Jar

jeremychiaaaaa committed:
Add JavaDoc

jeremychiaaaaa committed:
Add A-CodingStandard

jeremychiaaaaa committed:
Level-9

jeremychiaaaaa committed:
Merge branch 'branch-A-CodingStandard'

jeremychiaaaaa committed:
Merge tag 'Level-9'

# Conflicts:
#	src/main/java/NextGPT/Parser.java
#	src/main/java/NextGPT/TaskList.java

jeremychiaaaaa committed:
Add Level-10

FionaQY commented on src/main/java/NextGPT/NextGPT.java:
```diff
@@ -0,0 +1,69 @@
+package NextGPT;
```
Names representing packages should be in all lower case.

FionaQY commented on src/main/java/NextGPT/Parser.java:
```diff
@@ -0,0 +1,92 @@
+package NextGPT;
+import NextGPT.command.*;
```
For CS2103T, wildcards are not advised

FionaQY commented on src/main/java/NextGPT/Parser.java:
```diff
@@ -0,0 +1,92 @@
+     * @return Command that contains the properties of user input.
+     * @throws NextGPTException If user input is invalid.
+     */
+    static Command parse(String fullCommand) throws NextGPTException{
```
Good job making parse also identify the dates and descriptions

FionaQY commented on src/main/java/NextGPT/Storage.java:
```diff
@@ -0,0 +1,74 @@
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            if (task instanceof Event) {
+                text += "E,";
+                text += task.isDone()? "X,":" ,";
+                text += task.getDescription() + ",";
+                text += ((Event) task).getStart() + ",";
+                text += ((Event) task).getEnd() + "\n";
+            } else if (task instanceof Deadline) {
+                text += "D,";
+                text += task.isDone()? "X,":" ,";
+                text += task.getDescription() + ",";
+                text += ((Deadline) task).getBy() + "\n";
+            } else {
+                text += "T,";
+                text += task.isDone()? "X,":" ,";
+                text += task.getDescription();
+                text += "\n";
```
You could also make the Task class itself have the method to give this string representation

FionaQY commented on src/main/java/NextGPT/task/Task.java:
```diff
@@ -0,0 +1,47 @@
+/**
+ * Class for tasks.
+ */
+public class Task {
```
You could make this class abstract

FionaQY reviewed (commented):
LGTM. Just some nits to fix.

yikjunxian commented on src/main/java/NextGPT/Storage.java:
```diff
@@ -0,0 +1,74 @@
+     * @param taskList Task list to be saved.
+     * @throws IOException If error occurs while saving.
+     */
+    public void add_to_memory(TaskList taskList) throws IOException {
```
Nice use of verbs to represent method to add list of tasks to memory! However underscores are more commonly used for test methods. You could rename the method to addToMemory using camelCase if you prefer.

yikjunxian commented on src/main/java/NextGPT/TaskList.java:
```diff
@@ -0,0 +1,109 @@
+ */
+public class TaskList {
+
+    ArrayList<Task> todo_list;
```
You may prefer to use camelCase for variables instead of underscores.

yikjunxian commented on src/main/java/NextGPT/Ui.java:
```diff
@@ -0,0 +1,118 @@
+     * @param listSize Size of task list.
+     */
+    public String addTask(Task task, int listSize) {
+        return " added: " +
```
You may prefer to break before the '+' operator.

yikjunxian commented on src/main/java/NextGPT/command/DeleteCommand.java:
```diff
@@ -0,0 +1,45 @@
+ * Subclass of Command that deletes tasks from task list
+ */
+public class DeleteCommand extends Command{
+    int index;
```
You may prefer to add an access modifier to your class variable.

yikjunxian reviewed (approved):
LGTM. Just a few nits to fix.

jeremychiaaaaa committed:
Add assert statements to several command methods

All command classes override an abstract execute function that takes in tasks, ui and storage as parameters.
Asserting that all 3 are null makes sure that there are no errors in the code.

jeremychiaaaaa committed:
* branch-A-CodeQuality:

Refactor package name into lowercase to follow Java naming convention and erase unused code.
In addition, related import statements are grouped together and seperated with line breaks for clarity

jeremychiaaaaa committed:
Merge pull request #1 from jeremychiaaaaa/branch-A-Assertions

Add assert statements to several command methods

jeremychiaaaaa committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/NextGPT/MainWindow.java
#	src/main/java/NextGPT/NextGPT.java
#	src/main/java/NextGPT/NextGPTException.java
#	src/main/java/NextGPT/Parser.java
#	src/main/java/NextGPT/TaskList.java
#	src/main/java/NextGPT/Ui.java
#	src/main/java/NextGPT/command/AddCommand.java
#	src/main/java/NextGPT/command/Command.java
#	src/main/java/NextGPT/command/DeleteCommand.java
#	src/main/java/NextGPT/command/EditCommand.java
#	src/main/java/NextGPT/command/ExitCommand.java
#	src/main/java/NextGPT/command/FindCommand.java
#	src/main/java/NextGPT/command/ListCommand.java
#	src/main/java/NextGPT/task/Deadline.java
#	src/main/java/NextGPT/task/Event.java
#	src/test/java/NextGPT/ParserTest.java
#	src/test/java/NextGPT/TaskListTest.java

jeremychiaaaaa committed:
Merge pull request #2 from jeremychiaaaaa/branch-A-CodeQuality

* branch-A-CodeQuality:

jeremychiaaaaa committed:
The chat bot is currently unable to show a list of events for a specific date.

Providing the ability to display the schedule for a chosen date would be beneficial to most users. This added feature would help them organize and plan their activities for the day, improving productivity and organization.

To use this function, users can type "schedule <date>", with the date formatted as yyyy-mm-dd (e.g., schedule 2024-09-01).

jeremychiaaaaa committed:
Change folder name

jeremychiaaaaa committed:
Delete src/main/java/NextGPT directory

jeremychiaaaaa committed:
Add UserGuide

jeremychiaaaaa committed:
Add ReadMe page

jeremychiaaaaa committed:
Update header comment of Parser File

jeremychiaaaaa committed:
Update header of Main File

jeremychiaaaaa committed:
Update header comment of exit method

jeremychiaaaaa committed:
Add header command for exit method

jeremychiaaaaa committed:
Add header for exit command in EditCommand class
